### PR TITLE
Clean up/21.31.5/bryman

### PR DESCRIPTION
**Pull request by Bryman**
Addresses some easily fixed issues.  There are a few more commits for the Window class that I excluded when I realized that appears to be auto generated. I can open a pull request for those if you'd like.

Let me know if there are any concerns, commits I should exclude, or changes I may make.

## Proposed Changes

  - Remove unnecessary throws in method signatures
  - Remove redundant interface access modifier
  - Remove redundant final modifier
  - Add final modifier to field
  - Remove raw usage of parameterized class

### CODE REVIEW

**Comment by Idrinth** [comment]
Will not have time to review this tonight. You're right, that most of the window class is generated, wouldn't want to touch that code

File: src/main/java/de/idrinth/waraddonclient/model/addon/ActualAddon.java [hunk 1]
```diff
@@ -152,7 +152,7 @@ public String getInstalled() {
      * @param data
      * @return String
      */
-    private final String getStringFromObject(String key, javax.json.JsonObject data) {
+    private String getStringFromObject(String key, javax.json.JsonObject data) {
```
Comment (Idrinth) [inline src/main/java/de/idrinth/waraddonclient/model/addon/ActualAddon.java:155]: while you're at it, please import `javax.json.JsonObject`

File: src/main/java/de/idrinth/waraddonclient/service/FileWatcher.java [hunk 2]
```diff
@@ -28,10 +29,12 @@ public FileWatcher(GuiAddonList addonList, BaseLogger logger, Config config) thr
...
-        WatchEvent.Kind[] modes = new WatchEvent.Kind[2];
-        modes[0] = StandardWatchEventKinds.ENTRY_CREATE;
-        modes[1] = StandardWatchEventKinds.ENTRY_MODIFY;
-        path.toPath().register(watcher, modes);
+        path.toPath()
+        .register(
+            watcher,
+            Arrays.asList(
+                    StandardWatchEventKinds.ENTRY_CREATE, StandardWatchEventKinds.ENTRY_MODIFY)
+                .toArray(new WatchEvent.Kind<?>[] {}));
```
Comment (Idrinth) [inline src/main/java/de/idrinth/waraddonclient/service/FileWatcher.java:37]: this actually looks worse than before, why was this changed?

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: src/main/java/de/idrinth/waraddonclient/service/FileWatcher.java (hunk 2 above)
Comment (Bryman) [reply src/main/java/de/idrinth/waraddonclient/service/FileWatcher.java:37]: The goal of this change was to first add the type parameter to `WatchEvent`.  I may have been overzealous and attempted to remove `modes` as an unnecessary variable assignment at the same time. I must not have been thinking clearly at the time, as the following is more inline with what I was going for:
```
        watcher = path.toPath().getFileSystem().newWatchService();
        path.toPath().register(watcher, StandardWatchEventKinds.ENTRY_CREATE, StandardWatchEventKinds.ENTRY_MODIFY);
```

However, if the preference is to keep it closer to the original structure, I can instead change it to:
```
        WatchEvent.Kind<?>[] modes = new WatchEvent.Kind[2];
        modes[0] = StandardWatchEventKinds.ENTRY_CREATE;
        modes[1] = StandardWatchEventKinds.ENTRY_MODIFY;
```

File: src/main/java/de/idrinth/waraddonclient/model/addon/ActualAddon.java (hunk 1 above)
Comment (Bryman) [reply src/main/java/de/idrinth/waraddonclient/model/addon/ActualAddon.java]: `javax.json.JsonObject` was already imported. I pushed commit `c9db6228a02ce20ff7a8d8ca4a20393b98c6775e` to replace fully qualified class names with just the class name for any already imported classes.

File: src/main/java/de/idrinth/waraddonclient/service/FileWatcher.java (hunk 2 above)
Comment (Idrinth) [reply src/main/java/de/idrinth/waraddonclient/service/FileWatcher.java:37]: both options are fine for me

Comment (Bryman) [reply src/main/java/de/idrinth/waraddonclient/service/FileWatcher.java]: pushed commit `7e86bf8801661baf7eb8945ecbb0aa0ac763a234` containing the first option.
